R/keypair.R

```diff
@@ -34,7 +34,30 @@ keypair_credentials <- function(
 
 # Generate a JWT that can be used for Snowflake "key-pair" authentication.
 generate_jwt <- function(account, user, private_key, iat = NULL, jti = NULL) {
-  key <- openssl::read_key(private_key)
+  # password callback that searches parent environments for private_key_file_pwd
+  # called by openssl only if the key is encrypted
+  password_cb <- function(prompt = "") {
```

Comment:
The documentation for `openssl::read_key()` states that `password` is either a string or a callback function. Is the string not appropriate? 

Is searching through the set of calling frames for a variable named `params` normal practice? Could `private_key_file_pwd` be provided from `snowflake_credentials()` to `keypair_credentials` and then `generate_jwt`?